### ISSUE TITLE
appimage of these five apps

### ISSUE THREAD

**Issue by zaxebo1** [probonopd/AppImageKit#243]
Yesterday only, i discovered appimage. I am really excited.
The following utilities which I use extensively, are not in the appimage gallery of apps. 
Can - providing of these five apps/apps-recipe - be kindly added to the roadmap :

VVIMP:  **cryptsetup,  latexmk (including the entire texlive), emacs**

OPTIONAL: dia, jabref (should include the entire openjdk contained in itself)

**Comment by PHPJosh** [comment]
in about a month or so i will be making alot of appimages of apps there are none for as I am make an OS centered around AppImageKit and will need as much software available to end users as possible and will maintain which ever ones i make until there are other maintainers or the Upstream developers start making appimages (which is what i am hoping for) I will add what ever ones i do make to the Appimage repo

**Comment by zaxebo1** [comment]
thanks.
judging with the level of happiness i have with AppImage downloads i made today, probably i forsee myself as **maintainer** of some appimages **in future**    :-) 

but, may be i will try going through docs for making AppImage **myself** in november, when i will have lots of time to invest for some learning.

**Comment by probonopd** [comment]
Although I do some sample apps on Bintray, I don't have the time to package every application on earth. Hence I kindly ask you to send AppImage packaging requests to the upstream authors of the respective software. 

Another way to get AppImages is to convert deb ingredients to AppImage. I am working on making this much easier. How would you like maintaining something like

```
app: Lantern

ingredients:
  dist: trusty
  sources: 
    - deb http://archive.ubuntu.com/ubuntu/ trusty main universe
  debs:
    - https://s3.amazonaws.com/lantern/lantern-installer-beta-64-bit.deb
```

and get a working, up-to-date, AppImage?

**Comment by probonopd** [comment]
* https://bintray.com/probono/AppImages/Dia#files

**Comment by PHPJosh** [comment]
think this what @chrisnharvey made https://github.com/eloquentstore/appimager or something like it which is what your sample code looks similar to will help put more appimages out from quite a few of us. I haven't really have had the time to play with it yet but plan to once i get done compiling libs and other packages so i can make my first ISO to test them on once i fully fork the packages I need for the OS so Updates from Arch doesn't screw me up lol.. had something break this morn because  about 30% of the packages in Arch are broken because they have never updated them for the updated GCC 6 lol... which is why I am not making my OS a Rolling release.. seems alot gets lost in a rolling release.

But anyway I plan on  putting out a bit of AppImages once i get to the point of building ISOs so there is plenty of Apps to test on it, and any tool to help in that area will help us all until more upstream devs start making them.. which I am kinda hoping with an OS being built based on them gives them more incentive to do so :)

**Comment by probonopd** [comment]
Actually better help upstream application projects to roll official AppImages, like Krita, MuseScore, KDevelop, OpenShot, and others :-)

**Comment by PHPJosh** [comment]
yep any help given upstream will surely help them start pushing more appimages out

**Comment by probonopd** [comment]
And the good thing is, they know their software best, so these upstream packaged AppImages tend to work much, much better than anything we can make. Best example: the Krita and KDevelop AppImages. I would never have been able to do them with this level of polish.

**Comment by zaxebo1** [comment]
@probonopd  regarding your comment at https://github.com/probonopd/AppImageKit/issues/243#issuecomment-249925049 , it will be a good tool

===

As you have suggested to generate AppImages myself or help upstream in doing so, I appreciate your point for that can you answer me following ?

1) On which OS distribution and which version of that OS, you are generating those AppImages available on bintray??

2) just like you have appimage helper tool for Qt applications at https://github.com/probonopd/linuxdeployqt , in the same way do you have some appimage helper tool for Gtk apps too?

3) I am very enthusiastic for AppImage, and will try to package/help to package - many apps in future. But emacs is such a core app for many of us, that if you helped even the emacs appimage recipe at https://github.com/probonopd/AppImages/issues/94 , that will give a huge morale boost. Emacs is just an OS in itself for us. Emacs contains inbuilt visual display of org-mode, which is core of everything for many many of us.
With all the experience you have in appimages, if you helped in perfecting recipe of **even just one single software emacs** , that will be godsent for us. **We will be happily able to write/tweak/package/maintain other apps, but you just give your best help to just perfect emacs(which includes org-mode) recipe** . Please.  Please  Please   :-)  :-)

**Comment by probonopd** [comment]
> On which OS distribution and which version of that OS, you are yourself generating those AppImages available on bintray

I am using Ubuntu trusty on Travis CI for the majority of my example AppImages which means they will not run on distributions older than 2014-ish. This is because for many AppImages I don't compile the contained software from scratch (I don't have the time to do this) but have to use existing packages; and experience shows that it is hard to find halfway recent packages for distributions older than trusty.

However, if an upstream project is serious about providing AppImages (e.g., Krita), then I recommend them to compile the ingredients on debian oldstable or on CentOS 6 since the resulting binaries will be compatible to older systems this way.

>  do you have some appimage helper tool for Gtk apps too

Shhh, don't tell anyone, but linuxdeployqt has a command line switch to bundle non-Qt libraries, too... so it should work for other apps too.

> You didnot gave any link regarding jabref

Because it is not finished yet. Here is the preliminary recipe:
https://github.com/probonopd/AppImages/blob/master/recipes/meta/JabRef.yml

One done, it will appear on
https://bintray.com/probono/AppImages/JabRef

> can you give some url of recipe of AppImage of java(openjre) based software, which bundles openjre within it too

Check [these recipes](https://github.com/probonopd/AppImages/search?utf8=%E2%9C%93&q=JRE) in the probonopd/AppImages project (where I keep the recipes for my example AppImages).

> Emacs

I respect that this is an important piece of software for many people, but for me it is not user-friendly at all and seems to come out of another century. It does not follow today's basic UI design guidelines and looks like, as you correctly write, an OS inside the OS. This makes it a) me not want to invest time into it and b) makes it incredibly hard to package (not just for an AppImage but in general) - probably one of the hardest pieces of software there are. It takes someone who knows the software, and can possibly modify the software, to make it relocateable (i.e. make it run from changing locations at runtime, not just `usr` that was compiled in as `$PREFIX` at compilation time) alone (this is just an example). Philosophically, I get the impression that the GNU project does not care about binary compatibility, they just tell their users "get the source and compile it yourself". And apparently this is sufficient for their core audience, so I suspect there would not even be many users interested in an AppImage of it.

**Comment by zaxebo1** [comment]
thanks for all the tips.  

Regarding the use case of emacs appimage vis-a-vis compiling it: Though emacs users may be hardcore, as you said. But my use case is different:
My idea is that as I am heavy user of emacs, whenever I travel to my friends houses, they generally do not have emacs installed on their linux(as they may use vim/gedit only), or some friends have their emacs's old versions as they use CentOS 6 and i use debian testing and ubuntu.
I do not want to depend upon internet to download various dependencies to compile the latest emacs itself, i will like the idea of emacs portable single binary sitting in my pen drive, which i can use on my friends linux PCs anytime, and i know that it will behave as I want, because it is latest version , along with my addons in the recipe.
Basically with emacs appimage copied on my pen drive, i will feel psychologically very secure :-)  while going anywhere, even without my laptop.

**Comment by zaxebo1** [comment]
as a side note:

regarding "emacs is OS" the joke was that:
**Emacs is “a great operating system, lacking only a decent editor”**

but now with evil addon on emacs, it is jokingly said that:
      **Emacs OS has finally got its decent text editor too**

**Comment by probonopd** [comment]
Emacs seems to be working with some binary patching and this custom AppRun script:
https://github.com/probonopd/AppImages/blob/master/recipes/meta/Emacs.AppRun

Recipe:
https://github.com/probonopd/AppImages/blob/master/recipes/meta/Emacs.yml

AppImage (needs testing and refinement):
https://bintray.com/probono/AppImages/Emacs/_latestVersion#files

**Comment by zaxebo1** [comment]
I was away from town for sometime and today morning after getting back to home,  i got this AppImage of emacs as mentioned above. Thanks for the effort. 
as mentioned above, i tested it and giving the feedback below.

I downloaded the AppImage and ran it on ubuntu 16.04. When ran , it gives the following error message:

======

user1@machine-name1:~$   ./Emacs-2015072515+5e63841f2847b0b07cf15aed98c26fe914e9804a.glibc2.11-x86_64.AppImage 

ls: cannot access 'usr/lib/emacs': No such file or directory
/bin/bash: /tmp/.mount_Savle4/lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)
/tmp/.mount_Savle4/usr/bin/emacs-snapshot: /tmp/.mount_Savle4/lib/x86_64-linux-gnu/libgpg-error.so.0: no version information available (required by /lib/x86_64-linux-gnu/libgcrypt.so.20)
Warning: arch-independent data dir '/tmp/.mount_Savle4/usr/share/emacs///etc/': No such file or directory
Warning: Lisp directory '/tmp/.mount_Savle4/usr/share/emacs///lisp': No such file or directory
Warning: Lisp directory '/tmp/.mount_Savle4/usr/share/emacs///lisp/emacs-lisp': No such file or directory
Error: charsets directory not found:
/tmp/.mount_Savle4/usr/share/emacs/etc/charsets
Emacs will not function correctly without the character map files.
The EMACSDATA environment variable is set, maybe it has the wrong value?
Please check your installation!

===========

**Comment by zaxebo1** [comment]
replied the above at my comment url https://github.com/probonopd/AppImages/issues/94#issuecomment-250947915

**Comment by probonopd** [comment]
Try Emacs from https://github.com/probonopd/Emacs.AppImage
